### DISCUSSION TITLE
`import` statement does not work

### DISCUSSION THREAD

**Discussion by olaven** [vercel/next.js#13613] [Help]
Hi! I am building a site with dynamic API routes. These routes connect to a database using Typeorm. The server crashes when I use import statements, like this: 
```
/path/to/entity.ts
import "reflect-metadata"
       ^^^^^^^^^^^^^^^^^^

SyntaxError: Unexpected string
```

I am not using a custom server, and my understanding is that NextJS would transpile the code so that `require`-statements are not necessary. 
Does anyone know of an explanation? Thanks for any help! 😄 

Olav

**Comment by rafaelalmeidatk** [comment]
Hello! Can you share a repo with a minimal reproduction? This line works fine with the [typescript example](https://github.com/vercel/next.js/tree/canary/examples/with-typescript)

**Reply by olaven** [reply]
Sure! https://github.com/olaven/error-reproductions/tree/master/nextjs-import-error

**Reply by rafaelalmeidatk** [reply]
I tested the project here, the file `database/GetRepository.ts` is compiling but there is a typo [here](https://github.com/olaven/error-reproductions/blob/master/nextjs-import-error/pages/api/data.ts#L1), `getRepository` should be `GetRepository` (to match the same casing of the filename)

**Reply by olaven** [reply]
Thanks :-) I did not spot that as the project does not compile on my machine. 

What versions of Node are you on? (Or is there something else that could play in here?)

**Reply by rafaelalmeidatk** [reply]
Node v12.17.0. But the project isn't completely compiling because there is a type error in the `MyEntity` file:
```
ERROR in /tmp/error-reproductions/nextjs-import-error/database/MyEntity.ts(3,1):
3:1 Unable to resolve signature of class decorator when called as an expression.
  Type 'ClassDecorator' is not assignable to type 'typeof MyEntity'.
    Type 'ClassDecorator' provides no match for the signature 'new (): MyEntity'.
    1 | import {Entity, PrimaryGeneratedColumn, Column } from "typeorm";
    2 | 
  > 3 | @Entity 
      | ^
    4 | export class MyEntity {
    5 | 
    6 |     @PrimaryGeneratedColumn
```

**Reply by olaven** [reply]
This is strange. I don't have access to my laptop at the moment, but I'll investigate further and get back to you when i do. Thanks again!

**Reply by olaven** [reply]
Also Node v12.17.0, still getting 
```
import {Entity, PrimaryGeneratedColumn, Column } from "typeorm";
^^^^^^

SyntaxError: Cannot use import statement outside a module
```
